lib/active_model_serializers/adapter/attributes.rb

```diff
@@ -5,7 +5,11 @@ module Adapter
     class Attributes < Base
       def initialize(*)
         super
-        instance_options[:fieldset] ||= ActiveModel::Serializer::Fieldset.new(fields_to_fieldset(instance_options.delete(:fields)))
+
+        instance_options[:fieldset] ||= begin
```

Comment:
instance_options[:fieldset] could be:
1. the `fieldset` passed in through option
2. `ActiveModel::Serializer::Fieldset` instance
3. nil

Reply:
That's fine. Or add a class method on fieldset which does what's in this begin block

Reply:
echoing what you previously mentioned, I also want keep the changes minimal 🙏 
if it's necessary, is it better to do it in another refactoring PR?